app/src/main/java/io/awaken/ui/connections/ConnectionViewHolder.kt

```diff
@@ -0,0 +1,102 @@
+    internal val mac = ViewCompat.requireViewById<TextView>(itemView, R.id.mac_label)
+    internal val location = ViewCompat.requireViewById<TextView>(itemView, R.id.location_label)
+    internal val date = ViewCompat.requireViewById<TextView>(itemView, R.id.awoken_date_label)
+    internal val status = ViewCompat.requireViewById<ImageView>(itemView, R.id.status_marker)
+
+    private var connectionId: Int = 0
+    private val databaseHelper: ConnectionDatabaseHelper
+
+    init {
+
+        val editButton = ViewCompat.requireViewById<ImageButton>(itemView, R.id.edit_button)
+        val deleteButton = ViewCompat.requireViewById<ImageButton>(itemView, R.id.delete_button)
+
+        databaseHelper = ConnectionDatabaseHelper(view.context)
+
+        view.setOnClickListener { itemView ->
+            Wake.sendPacket(host.text.toString(), mac.text.toString())
+                    .subscribeOn(Schedulers.io())
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .subscribe {
+                        val dateFormat = SimpleDateFormat(DATE_FORMAT, Locale.US)
+                        val date = Date()
+                        val formatDate = dateFormat.format(date)
+                        databaseHelper.updateDate(connectionId, formatDate)
+                        recyclerView.setIsAnimatable(false)
+                        this.date.text = formatDate
+                        Snackbar.make(itemView, nickname.text.toString() + " "
+                                + itemView.resources.getString(R.string.woken),
+                                Snackbar.LENGTH_LONG).show()
+                    }
+        }
+
+        view.setOnLongClickListener { _ -> true }
+
+        editButton.setOnClickListener { itemView ->
+            val mainIntent = Intent(itemView.context, NewConnectionActivity::class.java)
+            itemView.context.startActivity(mainIntent)
+        }
+
+        deleteButton.setOnClickListener { itemView ->
+            val builder = AlertDialog.Builder(itemView.context, R.style.AlertDialog)
+            builder.setMessage("Are you sure you want to delete " + nickname.text.toString() + "?")
+            builder.setNegativeButton(android.R.string.cancel, null)
+            builder.setPositiveButton(android.R.string.ok) { dialog, which ->
+                var message: String
+                try {
+                    databaseHelper.deleteConnection(connectionId)
+                    message = "Successfully deleted " + nickname.text.toString()
+                    mRefresher.refreshConnections()
+                } catch (e: Exception) {
+                    message = "Failed to delete " + nickname.text.toString()
+                    Log.e("failure", e.message)
+                }
+
+                Toast.makeText(itemView.context, message, Toast.LENGTH_LONG).show()
+            }
+            builder.create()
+            builder.show()
+        }
+    }
+
+    fun setId(id: Int) {
```

Comment:
This function can be deleted and we can set `connectionId` directly in the adapter.